### ISSUE TITLE
Importing this package into CJS project triggers false-positive TypeScript error

### ISSUE THREAD

**Issue by grabbou** [google/zx#871]
### Expected Behavior

The following should work without any errors:

```js
import { $ } from 'zx'
```

### Actual Behavior
Triggers the following error:
```
The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("zx")' call instead.
  To convert this file to an ECMAScript module, change its file extension to '.mts', or add the field `"type": "module"` to 'package.json'.ts(1479)
```

However, the project works just fine. This is because `main` in `package.json` is set to a CJS entry-point (thank you!)

The only alternative I am aware of is setting `{type: module}` inside `esm` build directory. It would be great if `zx` could be used in both ESM and CJS projects without warnings.

**Comment by antongolub** [comment]
@grabbou,

Could you provide a full use case for debug?
* zx version
* platform
* reproducible setup
  * index.ts
  * package.json
  * tsconfig.json
  * tsc build params

**Comment by grabbou** [comment]
I would be happy to look into this and provide a proposed solution, once approved, submit a follow up PR. @antongolub et al, please let me know if you would consider a PR that fixes this.

**Comment by antongolub** [comment]
@grabbou,

This should be fixed in [8.1.5](https://github.com/google/zx/releases/tag/8.1.5). Could you check?
